integrations/spark/spark-3.5/openhouse-spark-runtime/src/main/scala/com/linkedin/openhouse/spark/sql/execution/datasources/v2/GrantRevokeStatementExec.scala

```diff
@@ -29,9 +30,9 @@ case class GrantRevokeStatementExec(
       case grantRevokableCatalog: SupportsGrantRevoke =>
         resourceType match {
           case GrantableResourceTypes.TABLE =>
-            grantRevokableCatalog.updateTableAclPolicies(Spark3Util.identifierToTableIdentifier(ident), isGrant, privilege, principal)
+            grantRevokableCatalog.updateTableAclPolicies(Spark3Util.identifierToTableIdentifier(ident), isGrant, privilege, Principal(principal))
           case GrantableResourceTypes.DATABASE =>
-            grantRevokableCatalog.updateDatabaseAclPolicies(toNamespace(ident), isGrant, privilege, principal)
+            grantRevokableCatalog.updateDatabaseAclPolicies(toNamespace(ident), isGrant, privilege, Principal(principal))
```

Comment:
Any reason why we have this only for the spark sql specifically?

Reply:
At least when taking into account of the scope of our clients, I'm treating this feature as an extension of the Spark SQL syntax defined in 
https://github.com/linkedin/openhouse/blob/c0190152e8383c31346e39d39876ac9a4af8f4b9/integrations/spark/spark-3.1/openhouse-spark-runtime/src/main/antlr/com/linkedin/openhouse/spark/sql/catalyst/parser/extensions/OpenhouseSqlExtensions.g4#L4

The reason why Principal isn't mapped from the sql extensions is to support the reverse mapping of `*` to `PUBLIC` when performing the `SHOW GRANTS` statements.

We can move this to server side but my thought was to have this handled like a keyword in the SQL syntax rather than an implicit keyword handled by the server.

Reply:
discussed offline, we currently have no plans to open this api beyond spark sql, lets revisit this when we have the requirement, thanks